### PR TITLE
Chore/ddw 249 improve build time

### PR DESCRIPTION
**Pull request by DominikGuzei**
**NOTE: PLEASE MERGE #886 before, since this PR is based on it**

This PR dramatically reduces the (successive) build time with a [proper caching solution](https://github.com/mzgoddard/hard-source-webpack-plugin).

The first build still takes the same amount of time as before (obviously), but the cache reduces it down to a few seconds for re-builds. However the cache is destroyed and rebuilt (automatically) if anything in the setup changes (e.g: package.json).

This improves the dev experience greatly, as you don't have to wait 40 seconds for each start of the app or running the tests if all you changed is a few lines of code.

## Todo:

- [x] Test the build on all platforms (_caching is disabled for builds_)
- [x] Double check if there are any issues with the caching during development / testing.

## Screenshots:

The build speed improvements in one pic:
screenshot 2018-05-09 um 15 00 12

---

### Basics

- [ ] PR is updated to the most recent version of target branch (and there are no conflicts)
- [ ] PR has good description that summarizes all changes and shows some screenshots or animated GIFs of important UI changes
- [ ] CHANGELOG entry has been added and is linked to the correct PR on GitHub
- [ ] Automated tests: All acceptance tests are passing (`npm run test`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *development* build (`npm run dev`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *production* build (`npm run package` / CI builds)
- [ ] There are no *flow* errors or warnings (`npm run flow-test`)
- [ ] There are no *lint* errors or warnings (`npm run lint`)
- [ ] Text changes are proofread and approved (Jane Wild)
- [ ] There are no missing translations (running `npm run manage-translations` produces no changes)
- [ ] UI changes look good in all themes (Alexander Rukin)
- [ ] Storybook works and no stories are broken (`npm run storybook`)
- [ ] In case of npm dependency changes both `package-lock.json` and `yarn.lock` files are updated

### Code Quality
- [ ] Important parts of the code are properly documented and commented
- [ ] Code is properly typed with flow
- [ ] React components are split-up enough to avoid unnecessary re-rendering
- [ ] Any code that only works in Electron is neatly separated from components

### Testing
- [ ] New feature / change is covered by acceptance tests
- [ ] All existing acceptance tests are still up-to-date
- [ ] New feature / change is covered by Daedalus Testing scenario
- [ ] All existing Daedalus Testing scenarios are still up-to-date

### After Review:
- [ ] Merge PR
- [ ] Delete source branch
- [ ] Move ticket to `done` on the Youtrack board

### CODE REVIEW

**Comment by DominikGuzei** [comment]
@nikolaglumac this is ready for review again!
As this greatly improves developer experience and happiness, i would suggest to review and merge sooner than later 😉

**Comment by DominikGuzei** [comment]
@nikolaglumac this PR doesn't change anything related to the translations

**Comment by nikolaglumac** [comment]
Are you sure @DominikGuzei? I used your branch - made a couple of changes and then ran `npm run manage-translations` and all of the used keys were deleted :(

**Comment by DominikGuzei** [comment]
You can (and should have) see the code changes 😆 nothing at all is related to the translations

**Comment by nikolaglumac** [comment]
@DominikGuzei this is what I am seeing:
1. delete `daedalus/translations/messages/` folder contents 
2. run `npm run dev`
3. this should result in `source/` folder being added in `daedalus/translations/messages/`
If I do this on `develop` branch it works fine but if I do it on your branch nothing get's created :(
I know what is the part that caused this problem and I will direct you to it.

File: source/renderer/webpack.config.js [hunk 1]
```diff
@@ -24,8 +29,7 @@ module.exports = {
         include: /source/,
         exclude: /source\/main/,
         use: {
-          loader: 'babel-loader?cacheDirectory&cacheIdentifier=' + Math.random(),
-          // Fix for https://github.com/yahoo/babel-plugin-react-intl/issues/47 ^^
```
Comment (nikolaglumac) [inline source/renderer/webpack.config.js]: @DominikGuzei this is the part which broke the intl setup...

Comment (nikolaglumac) [reply source/renderer/webpack.config.js]: Perhaps we could put this as an integral part of the ‘npm run manage-translations’ task?

Comment (DominikGuzei) [reply source/renderer/webpack.config.js]: I just pushed a fix: translations are never touched except when you run `npm run manage-translations` (then they are purged, cache cleared, build + regenerated)

Comment (nikolaglumac) [reply source/renderer/webpack.config.js]: Thanks Dominik!
Will check it out in the morning ;)

Comment (nikolaglumac) [reply source/renderer/webpack.config.js]: Thanks Dominik!
Will check it out in the morning ;)

Comment (nikolaglumac) [reply source/renderer/webpack.config.js]: Works like charm 👍

**Comment by nikolaglumac** [comment]
@DominikGuzei this one is ready to be merged as soon as the CI is done. Great work! 🎉

**Pull request merged**